Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -0,0 +1,75 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using AzureCognitiveSearch.PowerSkills.Common;
+using Newtonsoft.Json.Linq;
+
+namespace AzureCognitiveSearch.PowerSkills.Text.CustomEntitySearch
+{
+    /// <summary>
+    /// Based on sample custom skill provided in Azure Search. Provided a user-defined list of entities
+    /// this function determines the Entity first occurrence within a given document. This list of entities
+    /// must repeatedly be provided by the user for each document.
+    /// </summary>
+    public static class CustomEntitySearch
+    {
+        private static int MaxRegexEvalTime = 1;
+        /// <summary>
+        /// We assert the following assumptions:
+        /// 1. All text files are Latin based (i.e. no languageCode)
+        /// 2. Words can contain special characters and numbers
+        /// 3. The provided entities are not case sensitive
+        /// </summary>
+
+        [FunctionName("custom-search")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log, ExecutionContext executionContext)
+        {
+            log.LogInformation("Custom Entity Search function: C# HTTP trigger function processed a request.");
+
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
+                (inRecord, outRecord) => {
+                    string text = inRecord.Data["text"] as string;
+                    List<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
+
+                    List<Entity> data = new List<Entity>();
+                    if (!string.IsNullOrWhiteSpace(text))
+                    {
+                        foreach (string word in words)
+                        {
+                            if (string.IsNullOrEmpty(word)) continue;
+                            string escapedWord = Regex.Escape(word);
+                            string pattern = @"\b(?ix:" + escapedWord + ")";
+                            Match entityMatch = Regex.Match(text, pattern, RegexOptions.IgnoreCase, TimeSpan.FromSeconds(MaxRegexEvalTime));
+                            data.Add(
+                                new Entity
+                                {
+                                    Name = word,
+                                    MatchIndex = entityMatch.Success ? entityMatch.Index : -1
+                                }) ;
+                        }
+                    }
+
+                    outRecord.Data["Entity"] = data;
```

Comment:
>Entity [](start = 36, length = 6)

this one i think you would name plural since its a list. Entities

Reply:
Yes, and the `data` variable could be more expressively named `entities` as well.